### PR TITLE
feat(integrations): Implement fetching create issue fields for JIRA

### PR DESCRIPTION
**Pull request by macqueen**
a lot of this is copied from sentry plugins and just adjusted slightly

cc @MeredithAnya @lauryndbrown

### CODE REVIEW

File: src/sentry/integrations/jira/integration.py [hunk 1]
```diff
@@ -48,6 +58,167 @@ def create_comment(self, issue_id, comment):
...
+        }
+        # override defaults based on field configuration
+        if (schema['type'] in ['securitylevel', 'priority']
+                or schema.get('custom') == JIRA_CUSTOM_FIELD_TYPES['select']):
+            fieldtype = 'select'
+            fkwargs['choices'] = self.make_choices(field_meta.get('allowedValues'))
+        elif field_meta.get('autoCompleteUrl') and \
+                (schema.get('items') == 'user' or schema['type'] == 'user'):
+            pass
+            # TODO(jess): implement autocomplete for users
```
Comment (macqueen) [inline src/sentry/integrations/jira/integration.py]: actually implementing this is somewhat involved, so wanted to just do that in a separate PR

**Review by MaxBittker — APPROVED** [review]
this looks safe 👍
